core/capabilities/gateway_connector/service_wrapper.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
+)
+
+type serviceWrapper struct {
+	services.StateMachine
+
+	config    *config.GatewayConnector
+	keystore  keystore.Eth
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+}
+
+type workflowConnectorSigner struct {
+	services.StateMachine
+
+	connector   connector.GatewayConnector
+	signerKey   *ecdsa.PrivateKey
+	nodeAddress string
+	lggr        logger.Logger
+}
+
+var (
+	_ connector.Signer = &workflowConnectorSigner{}
+)
+
+func NewWorkflowConnectorSigner(config *config.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*workflowConnectorSigner, error) {
+	return &workflowConnectorSigner{
+		nodeAddress: (*config).NodeAddress(),
+		signerKey:   signerKey,
+		lggr:        lggr.Named("WorkflowConnectorSigner"),
+	}, nil
+}
+
+func (h *workflowConnectorSigner) Sign(data ...[]byte) ([]byte, error) {
+	return gwCommon.SignData(h.signerKey, data...)
+}
+
+func (h *workflowConnectorSigner) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+}
+func (h *workflowConnectorSigner) Start(ctx context.Context) error {
+	return h.StartOnce("WorkflowConnectorHandler", func() error {
+		return nil
+	})
+}
+func (h *workflowConnectorSigner) Close() error {
+	return h.StopOnce("WorkflowConnectorHandler", func() (err error) {
+		return nil
+	})
+}
+
+func (h *workflowConnectorSigner) SetConnector(connector connector.GatewayConnector) {
+	h.connector = connector
+}
+
+func translateConfigs(f config.GatewayConnector) connector.ConnectorConfig {
+	r := connector.ConnectorConfig{}
+	if f.NodeAddress() != "" {
+		r.NodeAddress = f.NodeAddress()
+	}
+
+	if f.DonID() != "" {
+		r.DonId = f.DonID()
+	}
+
+	if len(f.Gateways()) != 0 {
+		r.Gateways = make([]connector.ConnectorGatewayConfig, len(f.Gateways()))
+		for index, element := range f.Gateways() {
+			r.Gateways[index] = connector.ConnectorGatewayConfig{Id: element.ID(), URL: element.URL()}
+		}
+	}
+
+	if !reflect.ValueOf(f.WSHandshakeTimeoutMillis).IsZero() {
+		r.WsClientConfig = network.WebSocketClientConfig{HandshakeTimeoutMillis: f.WSHandshakeTimeoutMillis()}
+	}
+
+	// 0 are valid values
+	r.AuthMinChallengeLen = f.AuthMinChallengeLen()
+	r.AuthTimestampToleranceSec = f.AuthTimestampToleranceSec()
+	return r
+}
+
+// NOTE: this wrapper is needed to make sure that our services are started after Keystore.
+func NewGatewayConnectorServiceWrapper(config *gwConfig.GatewayConnector, keystore keystore.Eth, lggr logger.Logger) *serviceWrapper {
+	return &serviceWrapper{
+		config:   config,
+		keystore: keystore,
+		lggr:     lggr,
+	}
+}
+
+func (e *serviceWrapper) Start(ctx context.Context) error {
```

Comment:
Can you add a unit test that creates the ServiceWrapper object and then calls Start() can Close() on it, please?
You can take inspiration from functions/plugin_test.go and functions/connector_handler_test.go on how to mock the dependencies.

Ideally you'd test valid NodeAddress and an invalid one (i.e. key doesn't exit).